flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputGate.java

```diff
@@ -80,13 +80,17 @@
 	/**
 	 * Blocking call waiting for next {@link BufferOrEvent}.
 	 *
+	 * <p>Note: It should be guaranteed that the previous returned buffer has been recycled before getting next one.
```

Comment:
It is better to add some explanations here. 
.....next one, because the {@link BufferDecompressor} reuses the same intermediate buffer for storing the decompressed data.